server/src/main/scala/za/co/absa/atum/server/api/service/PartitioningServiceImpl.scala

```diff
@@ -38,7 +39,36 @@ class PartitioningServiceImpl(partitioningRepository: PartitioningRepository)
   ): IO[ServiceError, Either[StatusException, Unit]] = {
     repositoryCallWithStatus(
       partitioningRepository.createOrUpdateAdditionalData(additionalData), "createOrUpdateAdditionalData"
-    )
+    ).mapError(error => ServiceError(error.message))
+  }
+
+  override def getPartitioningMeasures(partitioning: PartitioningSubmitDTO): IO[ServiceError, Seq[MeasureDTO]] = {
+    partitioningRepository.getPartitioningMeasures(partitioning)
+      .mapError { case DatabaseError(message) =>
+        ServiceError(s"Failed to retrieve partitioning measures': $message")
+      }
+  }
+
+  override def getPartitioningAdditionalData(partitioning: PartitioningSubmitDTO): IO[ServiceError, AdditionalDataDTO] = {
+    partitioningRepository.getPartitioningAdditionalData(partitioning)
+      .mapError { case DatabaseError(message) =>
+        ServiceError(s"Failed to retrieve partitioning additional data': $message")
+      }
+  }
+
+  override def returnAtumContext(partitioningSubmitDTO: PartitioningSubmitDTO): IO[ServiceError, AtumContextDTO] = {
```

Comment:
I like how you named and implemented this function! However, few notes below

Reply:
I find it a bit confusing that the underlying DB function is called `createPartitioningIfNotExists` and all Scala related code around it as well, and then it's called here in `returnAtumContext` - and then `returnAtumContext` is again called in `createPartitioningIfNotExists` from `PartitioningControllerImpl` - so the naming createPartitioningIfNotExists is used on top level / API level as well as DB level - I don't know, I would perhaps try to distinguish them. What do you think?

(but I understand why it was done like this)

Reply:
I have now removed this function and implemented the functionality according to the old implementation.